dts/arm/nordic/nrf52833_qdaa.dtsi

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <nordic/nrf52833.dtsi>
+
+&flash0 {
+	reg = <0x00000000 DT_SIZE_K(512)>;
+};
+
+&sram0 {
+	reg = <0x20000000 DT_SIZE_K(128)>;
+};
+
+/ {
+	soc {
+		compatible = "nordic,nRF52833-QDAA", "nordic,nRF52833", "nordic,nRF52", "simple-bus";
```

Comment:
```suggestion
		compatible = "nordic,nrf52833-qdaa", "nordic,nrf52833", "nordic,nrf52", "simple-bus";
```

Reply:
Are you sure? that's not how we have it in the other SoCs today

Reply:
I've never seen compatibles using uppercase, but I don't see any rule in the dt spec

Reply:
@anangl your opinion here?

Reply:
In Devicetree Specification v0.4 (released 28 June) the following note was added in section "2.3.1 compatible":
> The compatible string should consist only of lowercase letters, digits and dashes, and should start with a letter.
A single comma is typically only used following a vendor prefix. Underscores should not be used.

I think we should correct those compatibles in all our SoCs then.